docs/xostor/xostor.md

```diff
@@ -852,3 +852,64 @@ Note: iptables config must also be modified to remove LINSTOR port rules (edit `
 For more info:
 - https://linbit.com/blog/linstors-auto-evict/
 - https://linbit.com/drbd-user-guide/linstor-guide-1_0-en/#s-linstor-auto-evict
+
+### How to-enable dm-cache?
+
+:::warning
+This feature is currently experimental and is not covered by XCP-ng support.
+:::
+
+On each host, create a new PV and VG using your cache devices:
+```
+pvcreate linstor_group_cache <CACHE_DEVICES>
+vgcreate linstor_group_cache <CACHE_DEVICES>
+```
+
+Using the controller, enable the cache:
+```
+# Make sure the primary resource group is configured with cache support.
+# Use `linstor storage-pool list` to verify the group to modify.
+# The group name starts with "xcp-sr-".
+linstor rg modify xcp-sr-linstor_group_thin_device --layer-list drbd,cache,storage
+
+# Enable cache on volume group.
+linstor vg set-property xcp-sr-linstor_group_thin_device 0 Cache/CachePool linstor_group_cache
+```
+
+:::tip
+You can list caches on a host using `dmsetup ls`. Also one important thing, a cache is only created on diskful resources.
+:::
+
+#### How to configure the cache size?
+
+By default, a VDI uses a cache size of 1% of its volume size. But it can be changed globally for all VDIs:
+```
+linstor vg set-property xcp-sr-linstor_group_thin_device 0 Cache/Cachesize <PERCENTAGE>
+```
+
+You can change this value globally or on a particular resource definition with:
+```
+linstor rd set-property <VOLUME_NAME> Cache/Cachesize <PERCENTAGE>
+```
+
+It's totally arbitrary. You can go up to 20-30% for for VMS with a high write rate. This should be enough to support a significant number of requests. 10% for solicited VMs. Between 1-5% for VMs with a few requests. You can use 100% if you want, for example for a database on a small VDI with a lot of queries.
+
+:::warning
+The use of snapshots can consume more memory than necessary due to VHD chains that are too long. It's advisable to limit their use except via XOA during backup processes.
+:::
+
+#### How to switch between read and read-write mode?
+
+Simply use:
+```
+linstor vg set-property xcp-sr-linstor_group_thin_device 0 Cache/OpMode <MODE>
+```
+
+By default `writethrough` mode is used. This mode is only useful to improve read performance.
```

Comment:
I'm not sure to understand the cache system: are reads already "cached" by design in LINSTOR? ie doesn't wait to read to another node but having the local disk speed for it. Last time I did a bench, this is what I had. So I am probably ignoring something obvious in there.

Reply:
DRBD can use the Linux system's buffers. But by design you don't have a cache: the read requests can be sent to the local host currently used and/or on another machine to accelerate parallel requests.

The use case of dm-cache here is to use HDDs (or slow SSDs) on the DRBD level and a faster device like an NVME to cache the most read blocks.